### DISCUSSION TITLE
Parse attribute arguments and contain them in AttributeData even if the attribute is not an attribute type

### DISCUSSION THREAD

**Discussion by Rekkonnect** [dotnet/roslyn#60381] [Ideas]
When the `AttributeClass` is an `ErrorType` with candidate reason `NotAnAttributeType`, the attribute type is properly resolved, and so are its constructors, but not recognized as a type that can be used as an attribute. `ConstructorArguments` will be empty, even if the constructor is provided with arguments.

This behavior is a gray area, as we're most likely dealing with a compilation error anyway, which case the generator should probably not respect. However, I've encountered a case where the actual error hides behind a `System.Runtime` reference hell, which should not really affect the actual compilation other than failing to bind to `System.Attribute`. The generator should not be affected by this kind of error, as the environment under which it works is closed to already working references.

I believe that despite `NotAnAttributeType` error reasons, `ConstructorArguments` should be properly populated if it successfully binds to a type, even if it isn't an attribute type, even if there are no `System.Runtime` errors.

**Comment by RikkiGibson** [comment]
This seems reasonable to me. IIRC, there is a pretty hard correspondence between `ConstructorArguments` and `AttributeConstructor.Parameters`--we ensure the lengths match, we reorder arguments in source to match the parameters, etc. I touched this pretty recently in #59750.

If making this change requires "loosening" that correspondence, i.e. allowing mismatch in scenarios where the `AttributeConstructor` is some kind of error symbol, we might have to do some poking around to ensure that any possible breaks introduced by the change are acceptable.

Tagging @dotnet/roslyn-compiler

**Reply by RikkiGibson** [reply]
Related: https://github.com/dotnet/roslyn/blob/6b1593b7d126beb1cd806052e5e69b590f0d9372/src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs#L302-L303
